### PR TITLE
Add threadpool implementation

### PR DESCRIPTION
**Pull request by uSpike**
This is pretty much a rewrite, but it's my attempt at a threadpool implementation.

Sharing from my own work on a different project.

### CODE REVIEW

File: anyio_serial/_impl.py [hunk 1]
```diff
@@ -2,84 +2,159 @@
...
+            partial(run_sync_in_worker_thread, self._read_worker, cancellable=True)
+        )
+        await self._task_group.spawn(
+            partial(run_sync_in_worker_thread, self._write_worker, cancellable=True)
+        )
+        return self
+
+    async def __aexit__(self, *args: Any) -> None:
+        await self.aclose()
+        await self._task_group.__aexit__(*args)
```
Comment (smurfix) [inline anyio_serial/_impl.py:75]: You should (a) wrap a shielded cancel scope around the `aclose` call, (b) return the value from `__aexit__`.

File: setup.cfg [hunk 2]
```diff
@@ -24,7 +24,7 @@ packages = anyio_serial
 python_requires = >= 3.6.2
 zip_safe = True
 install_requires =
-    anyio >= 2
+    anyio >= 2, <3
```
Comment (smurfix) [inline setup.cfg:27]: Given that anyio3 is right around the corner …

File: tests/test_serial.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    """
+    Test that all data will be received with receive() eventually
+    """
+    data = b"." * 100
+    recv = b""
+    async with serial:
+        await serial.send(data)
+        while True:
+            recv += await serial.receive()
+            if not serial.in_waiting:
```
Comment (smurfix) [inline tests/test_serial.py:46]: I'd rather break when `len(recv)==100`

File: setup.cfg (hunk 2 above)
Comment (uSpike) [reply setup.cfg:27]: Yeah, I'll update to anyio 3

File: anyio_serial/_impl.py (hunk 1 above)
Comment (uSpike) [reply anyio_serial/_impl.py:75]: `aclose()` does have a shielded cancel scope already, but I can move it to `__aexit__()`.

Comment (smurfix) [reply anyio_serial/_impl.py:75]: Huh. I seem to have completely missed that. No matter then.

**Pull request closed without merging**